tools-local/Microsoft.DotNet.Build.Tasks.Local/UploadClient.cs

```diff
@@ -142,6 +141,14 @@ public string EncodeBlockIds(int numberOfBlocks, int lengthOfId)
                     var req = new HttpRequestMessage(HttpMethod.Put, blockListUploadUrl);
                     req.Headers.Add(AzureHelper.DateHeaderString, dt1.ToString("R", CultureInfo.InvariantCulture));
                     req.Headers.Add(AzureHelper.VersionHeaderString, AzureHelper.StorageApiVersion);
+                    if(!string.IsNullOrEmpty(DetermineContentTypeBasedOnFileExtension(filePath)))
```

Comment:
It's a goal of mine, to get all of the tasks under Microsoft.DotNet.Build.Tasks.Local merged back into BuildTools where they can be consumed more general purpose by other repo's.  So, flexibility is slightly more important for these tasks than it would be for the tasks under `tools-local\tasks` which are more core-setup specific.